app/api/v1/views.py

```diff
@@ -0,0 +1,113 @@
+
+
+app = Flask(__name__)
+api = Api(app)
+
+
+class Users(Resource):
+    """This is the user class for signup"""
+    def post(self):
+        """The post request for the class"""
+        payload = api.payload
+        # check if payload is empty
+        if not payload:
+            return {'status': 'failed', 'message': 'provide all fields'}, 400
+        if payload:
+            if not (payload['email'] or payload['name'] or payload['password'] or payload['role']):
+                return {'status': 'failed', 'message': 'bad request refer to API document'}, 400
+            user = UserModel()
+            data = user.add_user(payload)
+            return {'status': 'added', 'message': 'Successfully signed up', 'data': data}, 201
+
+
+class Login(Resource):
+    """The login class that will handle login"""
+    def post(self):
+        """The post request for login"""
+        payload = api.payload
+        # Get the variables you'll use
+        email = payload['email']
+        page = payload['page']
+        password_given = payload['password']
+        user = UserModel()
+        data = user.get_user(email)
+
+        if not data: # returned from our model get_user
+            return {'status': 'failed', 'message': 'email not found'}, 401
+        role = data['role']
+        password = data['password']
+        if not password == password_given:
+            return {'status': 'failed', 'message': 'wrong password'}, 401
+        if page == 'admin' and role == 'user':
+            return {'status': 'failed', 'message': 'you are not an admin'}, 403
+        return {'status': 'success', 'message': 'redirect to ' + page}, 200
+
+
+class Parcels(Resource):
+    """The parcel class which handles parcels"""
+    def post(self):
+        """The post request from customer"""
+        if not api.payload:
+            return {'status': 'failed', 'message': 'please provide a json data'}, 400
+        payload = api.payload
+        if 'user_id' not in payload:
+            return {'status': 'failed', 'message': 'please provide user_id'}, 400
+        if 'parcel_type' not in payload:
+            return {'status': 'failed', 'message': 'please provide parcel_type'}, 400
+        if 'Dest' not in payload:
+            return {'status': 'failed', 'message': 'please provide destination'}, 400
+        if 'recepient_number' not in payload:
+            return {'status': 'failed', 'message': 'please provide receipt_number'}, 400
+        parcel = ParcelModel()
+        data = parcel.add_parcel(payload)
+        return {'status': 'added', 'message': 'Successfully added', 'parcel': data}, 201
+
+    def get(self):
+        """This gets all the parcels from the ParcelModel"""
+        parcel = ParcelModel()
+        parcels = parcel.get_all_parcels()
+        if not parcels:
+            return {'status': 'success', 'message': 'no parcel submited yet'}, 200
+        return {'status': 'success', 'data': parcels}, 200
+
+
+class GetOneParcel(Resource):
```

Comment:
Since the get GetOneParcel and put CancelParcel requests take an additional argument i.e. `order_no `, let's combine them into one Class that will contain both the `get` and `put` methods.

Reply:
Sure i will implement

Reply:
Hi Chris I tried putting them on one class but they are conflicting because of the different url endpoints , any advice on how i can tackle that

Reply:
Okay, makes sense if they are using different URLs.